### PR TITLE
Prime Ideals

### PR DESCRIPTION
**Pull request by tirix**
This updates a part of Jeff Madsen's section about prime ideals to a structure-based definition.
I've also added a few theorems of my own.

### CODE REVIEW

File: set.mm [hunk 1]
```diff
@@ -471539,6 +471657,281 @@ commutative monoid (=vectors) together with a semiring (=scalars) and a
...
+      XCTZYEWBUXRUWQLUYRUXHUWMBCUXGUXBYFVFYGUXMUYGLUYLLUXHUXKUXMUYGUYLUXLUXLGHU
+      WTUWTYHYIYJYKUYNUXIUVJUXJUVKUYNUVJUVDCPUYECPZUXIUYNUVHCUVDVUPYLUYNUVDUYEC
+      VUNYNUYNUXRUWQVUQUXIYOUYTVUBUWMBCUVPUXBYFTYMUYNUVKUVECPUYJCPZUXJUYNUVHCUV
+      EVUPYLUYNUVEUYJCVUOYNUYNUXRUWQVURUXJYOUYTVUBUWMBCUXEUXBYFTYMUUAUUBUYIUVEU
+      WTUWMUUCZJZUYKHUWTYPUYDVUTUYGUYFUYDUVEUVNVUSUYBUYAUVIXGUULVUSUVNPUUMUXTUY
+      AUVIUULUWMBAUWTUUFIVUDUULDXJUULUWTXKUULBCVOUUDUULUUEUUGYAZYQWHHUWTUVEUWMY
+      RRYSUYDUVDVUSJUYFGUWTYPUYDUVDUVNVUSUUNUXTUYAUVIYBVVAYQGUWTUVDUWMYRRYSXBUU
+      HUUIEFUVNAUVFUVHUWGVUMUWHUUJYTAUUKYT $.
+
+    $( An ideal ` I ` in the ring ` R ` is prime if and only if the factor ring
```
Comment (savask) [inline set.mm:471925]: Better to say "in the commutative ring", and similarly in some theorems above which assume `R e. CRing`.

**Comment by savask** [comment]
Nice work! Would be nice to formalize the fact that commutative rings with 1 have prime ideals. Jeff Madsen's work on maximal ideals might prove to be useful.

File: set.mm [hunk 2]
```diff
@@ -471236,6 +471248,84 @@ commutative monoid (=vectors) together with a semiring (=scalars) and a
...
+    $( The equivalence class of a cartesian product is the whole set.
+       (Contributed by Thierry Arnoux, 15-Jan-2024.) $)
+    ecxpid $p |- ( X e. A -> [ X ] ( A X. A ) = A ) $=
+      ( vx wcel cxp cec cv wbr cvv wb vex elecg mpan brxp baib bitrd eqrdv ) BA
+      DZCBAAEZFZARCGZTDZBUASHZUAADZUAIDRUBUCJCKUABSIALMUCRUDBUAAANOPQ $.
+  $}
+
+  ${
+    eqg0el.1 $e |- .~ = ( G ~QG H ) $.
+    $( Quotient group equivlance class of a subgroup.  (Contributed by Thierry
```
Comment (avekens) [inline set.mm:471262]: equivlance -> equivalence.

Maybe better "Equivalence class of a quotient group for a subgroup."

File: set.mm [hunk 3]
```diff
@@ -471236,6 +471248,84 @@ commutative monoid (=vectors) together with a semiring (=scalars) and a
...
+      ovexd id qusbas c0 wne grpbn0 qsxpid syl 3eqtr3d ) CFGZACAHIZJAAAKZJZBLMA
+      NZUJUKULAACDOPUJUKCBAQFBCUKRISUJETACLMSUJDTUJCAHUAUJUBUCUJAUDUEUMUNSACDUF
+      AUGUHUI $.
+  $}
+
+  ${
+    $d B x $.  $d G x $.  $d H x $.  $d Q x $.
+    qustrivr.1 $e |- B = ( Base ` G ) $.
+    qustrivr.2 $e |- Q = ( G /s ( G ~QG H ) ) $.
+    $( Reciproque of ~ qustriv .  (Contributed by Thierry Arnoux,
```
Comment (avekens) [inline set.mm:471318]: Is reciproque an English word (it's definitively French). Would "reciprocal" be an adequate translation?

File: set.mm [hunk 4]
```diff
@@ -471356,6 +471446,34 @@ commutative monoid (=vectors) together with a semiring (=scalars) and a
       CDXPXRVPVQVRWPYFEYCXAWIYCRWKYEWOWIYCCWJVSVTWAVQWPEXAWDWBWC $.
   $}
 
+  ${
+    $d .x. x $.  $d B x $.  $d I x $.  $d K x $.  $d R x $.  $d X x $.
+    rspsnel.1 $e |- B = ( Base ` R ) $.
+    rspsnel.2 $e |- .x. = ( .r ` R ) $.
+    rspsnel.3 $e |- K = ( RSpan ` R ) $.
+    $( Membership to a principal ideal.  Analogous to ~ lspsnel .  (Contributed
```
Comment (avekens) [inline set.mm:471454]: "Membership in a principal ideal."

File: set.mm [hunk 5]
```diff
@@ -471539,6 +471657,281 @@ commutative monoid (=vectors) together with a semiring (=scalars) and a
...
+       i =/= B /\ A. a e. ( LIdeal ` R ) A. b e. ( LIdeal ` R ) ( A. x e. a A.
+       y e. b ( x .x. y ) e. i -> ( a C_ i \/ b C_ i ) ) ) } ) $=
+      ( vr crg wcel cv cbs cfv cmulr wral clidl fveq2 wne co wo wi crab cprmidl
+      wss wa cvv df-prmidl wceq syl6eqr neeq2d eleq1d 2ralbidv imbi1d raleqbidv
+      oveqd anbi12d rabeqbidv id eqid fvexd rabexd fvmptd3 ) DLMZKDFNZKNZOPZUAZ
+      ANZBNZVHQPZUBZVGMZBHNZRAGNZRZVQVGUGVPVGUGUCZUDZHVHSPZRZGWARZUHZFWAUEVGCUA
+      ZVKVLEUBZVGMZBVPRAVQRZVSUDZHDSPZRZGWJRZUHZFWJUEZLUFUIABFKGHUJVHDUKZWDWMFW
+      AWJVHDSTZWOVJWEWCWLWOVICVGWOVIDOPCVHDOTIULUMWOWBWKGWAWJWPWOVTWIHWAWJWPWOV
+      RWHVSWOVOWGABVQVPWOVNWFVGWOVMEVKVLWOVMDQPEVHDQTJULURUNUOUPUQUQUSUTVFVAVFW
+      MFWJWNUIWNVBVFDSVCVDVE $.
```
Comment (avekens) [inline set.mm:471701]: empty line missing

File: set.mm [hunk 6]
```diff
@@ -471539,6 +471657,281 @@ commutative monoid (=vectors) together with a semiring (=scalars) and a
...
+      cmn4 ) BJKBLMZNKEAKFAKOGAKDAKOEFCPGDCPCPEGCPFDCPCPQBUDUDRZSACUDDEFGABUDUE
+      HTBCUDUEIUAUCUB $.
+  $}
+
+  ${
+    $d .x. m n r s $.  $d B m n r s x y $.  $d P m n r s x y $.
+    $d R m n r s x y $.
+    isprmidlc.1 $e |- B = ( Base ` R ) $.
+    isprmidlc.2 $e |- .x. = ( .r ` R ) $.
+    $( The predicate "is x prime ideal".  Alternate definition for commutative
```
Comment (avekens) [inline set.mm:471799]: Maybe better "The predicate "is prime ideal" for commutative rings.".

File: set.mm [hunk 7]
```diff
@@ -471539,6 +471657,281 @@ commutative monoid (=vectors) together with a semiring (=scalars) and a
...
+      IWJUVOXGYJMZYKXKXAYAUVQXTXAXQYAUVQXRCEXGYFGUUAWFSWKYJDXGWIWJWLWMWHWNWOWCW
+      DWRXAXQXOXBXRXOXQXDXEXNRZRXBXDXEXNWPXQXDUVRXBABCDEFGHWQWCWSSWT $.
+  $}
+
+  ${
+    $d I a y $.  $d I b $.  $d I e f x $.  $d I g h $.  $d Q a y $.  $d Q b $.
+    $d Q e f x $.  $d Q g h $.  $d R a b $.  $d R a y $.  $d R e f x y $.
+    $d R g h $.  $d e f g h $.
+    prmidlqusidom.1 $e |- Q = ( R /s ( R ~QG I ) ) $.
+    $( When the quotient ring of a ring is an integral domain, all its ideals
```
Comment (avekens) [inline set.mm:471836]: "When" -> "If"

File: set.mm (hunk 3 above)
Comment (tirix) [reply set.mm:471318]: I ended up using "converse", which I think is the correct term.

**Comment by tirix** [comment]
Thanks for all the remarks! 
Actually I find those theorems were [mentioned](https://github.com/metamath/set.mm/issues/1432#issuecomment-905806577) in the big survey made by @avekens and others about the "part 18". This might suggest those theorems are candidates to be [moved to main](https://github.com/metamath/set.mm/issues/1432#issuecomment-906617654), marking the original theorems as "Obsolete version of ~ xxx" as suggested by Norm at the time.

> Would be nice to formalize the fact that commutative rings with 1 have prime ideals. 

Yes, next I was looking at the proof that a commutative ring is an integral domain if and only if the zero ideal is prime.

File: set.mm [hunk 8]
```diff
@@ -471585,6 +471597,84 @@ commutative monoid (=vectors) together with a semiring (=scalars) and a
...
+    $( The equivalence class of a cartesian product is the whole set.
+       (Contributed by Thierry Arnoux, 15-Jan-2024.) $)
+    ecxpid $p |- ( X e. A -> [ X ] ( A X. A ) = A ) $=
+      ( vx wcel cxp cec cv wbr cvv wb vex elecg mpan brxp baib bitrd eqrdv ) BA
+      DZCBAAEZFZARCGZTDZBUASHZUAADZUAIDRUBUCJCKUABSIALMUCRUDBUAAANOPQ $.
+  $}
+
+  ${
+    eqg0el.1 $e |- .~ = ( G ~QG H ) $.
+    $( Equivalence class of a quotient group for a subgroup..  (Contributed by
```
Comment (avekens) [inline set.mm:471611]: ".." -> "."

**Review by avekens — APPROVED** [review]
Everything is fine now (except one small typo).

I would like to see this material in the main body of set.mm, because I think this is an important mathematical concept. Furthermore, ~df-ufd (unique factorization domains) is also in main, but currently based on PrIdl (of JM's mathbox, therefore commented out), and could be reactivated using PrmIdeal.

Are there additional theorems iu JM#s mathbox which could be "translated"? What about "Maximal ideals" (MaxIdl in JM's mathbox)? Is JM still active and could align his mathbox with TA's work?

**Comment by tirix** [comment]
> Everything is fine now (except one small typo).
> 
Fixed too!

> Is JM still active and could align his mathbox with TA's work?
>
It seems he is not.

He is a co-author of a [recent paper](https://www.researchgate.net/publication/358647215_Finiteness_of_associated_primes_of_local_cohomology_modules_over_Stanley-Reisner_rings)  (2022) about algebra, and an email address `madsen4@purdue.edu` is listed in the paper, but that address refused my messages. I've send a [mail to the list](https://groups.google.com/g/metamath/c/WnDkHHDtF4Y) just in case.

> What about "Maximal ideals" (MaxIdl in JM's mathbox)?

Yes, I'll have a look at them too.

**Review by savask — APPROVED** [review]

**Review by icecream17 — APPROVED** [review]
qsxpid seems simpler than qsalrel so I'll definitely check it out on the website
and the rest about prime ideals is great too

**Comment by tirix** [comment]
Oh, thanks for mentioning it, and sorry I did not find `qsalrel` when writing my proofs. 
I think I searched for `A X. A`, which does not appear as-is in your statement.

File: set.mm [hunk 9]
```diff
@@ -471888,6 +472006,283 @@ commutative monoid (=vectors) together with a semiring (=scalars) and a
...
+      EUFWBVSVNVMVPHDUHUIUJVIVQALPZQZWCDRZVSSZTZKVEQZWAKVEQLVEGWCGUDZWGWAKVEWIW
+      DVRWFVTVQAWCGUEWIWEVMVSWCGDUHUKUJUFVDWHLVEQZVHVDDVEMZDCULZWJVBVCWKWLWJURA
+      BCDEFLKIJUMUNUOUPVDVFVGUSUQVDVFVGUTUQVA $.
+
+    $d .x. a b $.  $d B a b x y $.  $d P a b x y $.  $d R a b x y $.
+    $( A condition that shows an ideal is prime.  For commutative rings, this
+       is often taken to be the definition.  See ~ ispridlc for the equivalence
+       in the commutative case.  (Contributed by Jeff Madsen, 19-Jun-2010.)
+       (Revised by Thierry Arnoux, 12-Jan-2024.) $)
+    prmidl2 $p |- ( ( ( R e. Ring /\ P e. ( LIdeal ` R ) ) /\ ( P =/= B /\ A.
```
Comment (avekens) [inline set.mm:472094]: Maybe we could add a reference to [Lang] p. 92 here.

Comment (avekens) [reply set.mm:472094]: Or maybe better at ~isprmidlc .

**Comment by avekens** [comment]
2 days without additional remarks/changes should be enough to merge this PR now.

**Pull request merged**